File: dsa-exercices-java/src/sort/BubbleSort.java
package sort;

import java.util.Arrays;

public class BubbleSort {

    public int[] sort(int[] input) {
        boolean isSorted = true;

        for (int k = 0; k < input.length; k++) {
            for (int i = 0; i < input.length - 1; i++) {
                int current = input[i];

                if (input[i] > input[i + 1]) {
                    isSorted = false;
                    input[i] = input[i + 1];
                    input[i + 1] = current;
                }
            }

            if (isSorted) {
                break;
            }
        }

        return input;
    }
}

File: dsa-exercices-javascript/sort/bubble-sort.js
function sort(input) {
    for (i = 0; i < input.length; i++) {
        let isSorted = true

        for (k = 0; k < input.length - 1; k++) {
            if (input[k] > input[k + 1]) {
                [input[k], input[k + 1]] = [input[k + 1], input[k]]
                isSorted = false
            }
        }

        if (isSorted) {
            break
        }
    }

    return input
}

const result = sort([1,2,3,4,5])
console.log(result)